### DISCUSSION TITLE
Calling a function with `queryClient.fetchQuery` inside of a hook `useQuery`

### DISCUSSION THREAD

**Discussion by WingsDevelopment** [TanStack/query#7708] [Q&A]
Is there any problem with calling a function with `queryClient.fetchQuery` inside of a hook `useQuery` ?

so what i want to do is 

```
const foo = () => {
 return queryClient.fetchQuery({ queryFN, queryKey });
}

const useBar = () => {
return useQuery({queryFn: () => foo(), queryKey});
}
```

it seems to me that it is working fine but I can't really find anything in documentation about this..

**Comment by TkDodo** [comment]
I don't understand what this indirection is supposed to do ...

**Comment by filipw01** [comment]
It used to work in v4, but in v5 I get this issue https://github.com/TanStack/query/issues/8494
